src/ethereum/ethTypes.ts

```diff
@@ -513,21 +513,37 @@ export interface EthPrivateKeys {
   mnemonic: string
   privateKey: string
 }
-export const asEthPrivateKeys = (pluginId: string): Cleaner<EthPrivateKeys> =>
-  asCodec(
+export const asEthPrivateKeys = (pluginId: string): Cleaner<EthPrivateKeys> => {
+  // Derived cleaners from the generic parameter:
+  const asFromKeys = asObject({
+    [`${pluginId}Mnemonic`]: asString,
+    [`${pluginId}Key`]: asString
+  })
+  const asFromJackedKeys = asObject({ keys: asFromKeys })
+  const asToKeys = asObject({
+    mnemonic: asString,
+    privateKey: asString
+  })
```

Comment:
This `asToKeys` isn't needed - TypeScript is fully aware that our return type is `EthPrivateKeys`, and will complain if we fail to match that expectation. You have already cleaned the raw input data, so that side is safe.

The rest of the changes here look good. I appreciate the humor in the variable naming.

Reply:
Ah, you're right. Got caried away with cleaners.

Time to decision is a valuable resource. Naming is hard; you made it easier. :)